internal/content_coding.go

```diff
@@ -133,19 +149,38 @@ type ContentEncoder interface {
 
 // GzipEncoder compresses the buffer using gzip at the default level.
 type GzipEncoder struct {
-	writer *gzip.Writer
-	buf    *bytes.Buffer
+	pwriter *pgzip.Writer
+	writer  *gzip.Writer
+	buf     *bytes.Buffer
 }
 
-func NewGzipEncoder() *GzipEncoder {
+func NewGzipEncoder(options ...EncodingOption) (*GzipEncoder, error) {
+	cfg := encoderConfig{level: pgzip.DefaultCompression}
+	for _, o := range options {
+		o(&cfg)
+	}
+
 	var buf bytes.Buffer
+	w, err := pgzip.NewWriterLevel(&buf, cfg.level)
 	return &GzipEncoder{
-		writer: gzip.NewWriter(&buf),
-		buf:    &buf,
-	}
+		pwriter: w,
+		writer:  gzip.NewWriter(&buf),
```

Comment:
```suggestion
        pw, err := pgzip.NewWriterLevel(&buf, cfg.level)
        if err != nil {
                return nil, err
        }
        w, err := gzip.NewWriterLevel(&buf, cfg.level)
        return &GzipEncoder{
                pwriter: pw,
                writer:  w,
```

@srebhan WDYT to change it to be like this?

Reply:
Agree.